data_loader.py

```diff
@@ -28,45 +31,103 @@ def disconnect(self):
         # Nothing to do
         pass
 
+    def _flatten_dict(self, data: dict, keystring: str = ""):
+        """Sufficent flattening dict"""
+        if type(data) == dict and len(data) != 0:
+            keystring = keystring + "." if keystring else keystring
+            for k in data:
+                yield from self._flatten_dict(data[k], keystring + str(k))
```

Comment:
Just fun fact. When we don't know the depth of the dictionary, it is better to use iteration over recursion.
Three advantages:
1) Iteration produces just one stack frame when recursion produces one stack frame for each function call
2) Iteration circumvent recursion limit in Python
3) Iteration in most cases is faster
If you would like to know more about it, you can look at my article https://www.przemodev.com/blog/articles/why-should-you-use-iteration-over-recursion :D